neurodamus/utils/memory.py

```diff
@@ -309,24 +311,45 @@ def collect_display_syn_counts(self):
         logging.info(" - Estimated synapse memory usage (MB):")
         from .logging import log_verbose
         inh_count = exc_count = 0
+        gap_count = other_count = 0
         log_verbose("+{:=^68}+".format(" Synapse Count "))
         log_verbose("| {:^40s} | {:^10s} | {:^10s} |".format("Synapse Type", "Family", "Count"))
         log_verbose("+{:-^68}+".format(""))
-        for syn_type, count in sorted(master_counter.items()):
+
+        # Some synapse types are numeric, others are strings, so we need to handle both
+        numeric_items = [(syn_type, count)
+                         for syn_type, count in master_counter.items()
+                         if isinstance(syn_type, (int, np.integer))]
+        for syn_type, count in sorted(numeric_items):
```

Comment:
Could you remind me why we sort "sorted(numeric_items)"?

Reply:
It's just for visualization, so when we print the table containing all the different types of synapses and their respective memory footprint they are ordered by their `syn_type_id`.

Reply:
If it is tolerant to print without sorting, I think we can fill the `inh/exc/gap/other_count` using one `for-loop`.  What is the usual size of `master_counter` ?

Reply:
It really depends on the circuit but normally we can expect in the most diverse circuits something in the order of ~40-50 different `syn_type_ids` i.e. `master_counter` keys to order. So the sorting is no big deal normally but if you think it's useless to order the synapses for visualization I can remove the sorting and make just one for loop, also removing the need for numeric vs string synapse types.

Reply:
OK, 40-50 is not a big size for loop, sort and then loop. We can stay like this ATM I think. 

BTW, I just pushed a commit to complete the `CONNECTION_TYPE` for NeuroModulationManager for the dry-run counting.